### PR TITLE
Use prism.js for code highlighting in guides [ci skip]

### PR DESCRIPTION
**Pull request by p8**
### Summary
SyntaxHighlighter is currently used in the guides for syntax highlighting. SyntaxHighlighter does not support highlighting for javascript and it seems to be no longer maintained.

Prism.js seems a nice alternative with support for Ruby, javascript, erb and Yaml.
It's used by Mozilla, Stripe, React and Drupal and uses the MIT License.
https://github.com/PrismJS/prism

### Before
image
image
image

### After
image
image
image

### CODE REVIEW

**Comment by simi** [comment]
Any chance to keep the original font?

**Comment by p8** [comment]
Yes, than it would look like this:
image

File: guides/assets/stylesheets/prism/default.css [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+	-moz-tab-size: 4;
+	-o-tab-size: 4;
+	tab-size: 4;
+
+	-webkit-hyphens: none;
+	-moz-hyphens: none;
+	-ms-hyphens: none;
+	hyphens: none;
+}
```
Comment (eileencodes) [inline guides/assets/stylesheets/prism/default.css]: The indentation in the other stylesheets are 2 spaces, not 4. Can you fix the indentation in these files?

File: guides/assets/stylesheets/prism/default.css [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+.token.italic {
+	font-style: italic;
+}
+
+.token.entity {
+	cursor: help;
+}
+
+
```
Comment (eileencodes) [inline guides/assets/stylesheets/prism/default.css]: Remove the extra lines at the bottom of the file.

**Review by eileencodes — CHANGES_REQUESTED** [review]
I definitely prefer the new style, it's a lot easier to read. There are some style issues in the files added to this PR, can you take a look at that?

File: guides/assets/stylesheets/prism/default.css (hunk 1 above)
Comment (p8) [reply guides/assets/stylesheets/prism/default.css]: Done!

File: guides/assets/stylesheets/prism/default.css (hunk 2 above)
Comment (p8) [reply guides/assets/stylesheets/prism/default.css]: Done!

**Review by eileencodes — APPROVED** [review]
This looks good to me, but I want to wait for @fxn to see if he has any strong opinions before merging 😄

**Comment by p8** [comment]
I've also improved the bash output:
image

**Comment by p8** [comment]
The only thing that doesn't seem to work correctly is the `end` keyword for `def new`
image

**Comment by fxn** [comment]
@eileencodes go ahead!

**Review by fxn — APPROVED** [review]

**Comment by p8** [comment]
I've created a pull request to fix the `def new end` formatting at the Prism repo.
https://github.com/PrismJS/prism/pull/2135/files
I can also apply the change to this pull request.

**Comment by kaspth** [comment]
@p8 Thanks so much! I think the improvement is better enough that we can live with that bug. Will you follow up for when your PR is merged/released? Just cc me on it.

**Comment by p8** [comment]
@kaspth Thanks! I'll let you know when the PR is merged.

**Comment by robin850** [comment]
Sorry I'm a bit late but why don't we highlight guides with [Rouge](https://github.com/rouge-ruby/rouge) ? This would avoid bundling a JavaScript dependency with guides.

**Comment by p8** [comment]
@robin850 Using Prism was an easy change, but I wouldn't mind using Rouge.

**Comment by kaspth** [comment]
Happy to see a spike using Rouge to get a feel for what that would be like if either of you are curious to investigate 🙌

**Comment by robin850** [comment]
@p8 : Yeah, that was just to speak about a potential alternative ; sorry if my message seemed a bit agressive / personal, that was not intended at all. :heart: 

If you want, you can look at using Rouge or I can during the holidays ; either way, it's fine.

**Comment by p8** [comment]
@robin850 No, worries ❤️. I'm taking it easy during the holidays, so I probably won't look at Rouge.

**Pull request merged**